### PR TITLE
138 display room names

### LINKED ISSUE #138: Display room names as labels
> Room names should be displayed as labels centered on rooms on the indoor map.

**Context**

- team BA will provide appropriate coordinates to display a room's label for each room when they implement the placement of markers on rooms

**Acceptance**

- [ ] every displayed room has a label showing its name as given by the database positioned according to what team BA provides
- [ ] labels are clickable and trigger an interface that will be defined by team BA and explained in this issue

### PR DESCRIPTION
**Pull request by dieknolle3333**
fixes #138

Run `rails runner app/helpers/import_gpx.rb importBuilding1.gpx`

the name of rooms is displayed in the middle of the room.
clicking on a room opens a pop-up that displays the name (but can contain any html).

### CODE REVIEW

File: app/views/map/_indoor.html.erb [hunk 1]
```diff
@@ -3,13 +3,33 @@
...
+            roomLayer = L.geoJSON(<%= raw(JSON.generate(room.to_geojson)) %>, 
+            {style: indoorStyle});
+            roomTooltip = L.tooltip({
+                permanent: true, 
+                interactive: true, 
+                className: 'marker_label', 
+                offset: L.point(0,0), 
+                direction: 'center',
+            });
+            roomTooltip.setContent("<%= room.full_name %>");
```
Comment (AbdullatifGhajar) [inline app/views/map/_indoor.html.erb:20]: If I am zoomed out, I see:
image
but if I'm zoomed in I see:
image

It has to be the opposite.

File: app/views/map/_indoor.html.erb [hunk 2]
```diff
@@ -3,13 +3,33 @@
...
         "color": "#444"
     };
+    let roomLayer;
+    let roomTooltip;
     <% @buildings.each do |building|%>
         <% building.rooms.each do |room|%>
+
+            roomLayer = L.geoJSON(<%= raw(JSON.generate(room.to_geojson)) %>, 
+            {style: indoorStyle});
+            roomTooltip = L.tooltip({
```
Comment (AbdullatifGhajar) [inline app/views/map/_indoor.html.erb:13]: this object can be extracted. As you can see in the source code, this object is duplicated 
image

File: app/views/map/_indoor.html.erb (hunk 1 above)
Comment (dieknolle3333) [reply app/views/map/_indoor.html.erb:20]: we did not deal with the zoom in this issue as https://github.com/hpi-swt2/compass-portal-red/issues/137 will deal with the display of the indoor data, including the room names.

File: app/views/map/_indoor.html.erb (hunk 2 above)
Comment (DevSchmidtchen) [reply app/views/map/_indoor.html.erb:13]: Are you referring to the full name of the room? So you propose to first store the Ruby variable `room.full_name` in a JavaScript variable and then use the JS variable from then on? We could definitely do that 👍

File: app/views/map/_indoor.html.erb (hunk 1 above)
Comment (dieknolle3333) [reply app/views/map/_indoor.html.erb:20]: update: having thought about it: at the moment, you can either see all names (including buildings and rooms) or none. if we switched this at this high level, you wouldn't be able to see the names of the buildings when zoomed out. is that really expected?
and splitting the display of room and building names up is actually a bit of work, that's why my call would be dealing with it in #137 and leaving it in as it is right now.

Comment (AbdullatifGhajar) [reply app/views/map/_indoor.html.erb]: Alright

**Review by AbdullatifGhajar — DISMISSED** [review]

**Review by DevSchmidtchen — APPROVED** [review]

**Review by DevSchmidtchen — APPROVED** [review]

**Pull request merged**